src/NuGet.Clients/VsExtension/NuGetPackage.cs

```diff
@@ -366,40 +364,93 @@ private void SetDefaultCredentialProvider()
             HttpHandlerResourceV3.CredentialsSuccessfullyUsed = (uri, credentials) =>
             {
                 // v2 stack credentials update
-                NuGet.CredentialStore.Instance.Add(uri, credentials);
+                CredentialStore.Instance.Add(uri, credentials);
             };
         }
 
-        private IEnumerable<NuGet.Credentials.ICredentialProvider> GetCredentialProviders()
+        private NuGet.Configuration.ICredentialService GetCredentialService()
         {
-            var packageSourceProvider = new PackageSourceProvider(new SettingsToLegacySettings(Settings));
+            // Initialize the credential providers.
             var credentialProviders = new List<NuGet.Credentials.ICredentialProvider>();
+            
+            TryAddCredentialProvider(
+                credentialProviders,
+                Resources.CredentialProviderFailed_LegacyCredentialProvider,
+                () =>
+                {
+                    var packageSourceProvider = new PackageSourceProvider(new SettingsToLegacySettings(Settings));
+
+                    return new CredentialProviderAdapter(new SettingsCredentialProvider(
+                        NullCredentialProvider.Instance,
+                        packageSourceProvider));
+                });
+            
+            TryAddCredentialProvider(
+                credentialProviders,
+                Resources.CredentialProviderFailed_VisualStudioAccountProvider,
+                () =>
+                {
+                    var importer = new VsCredentialProviderImporter(
+                        _dte,
+                        VisualStudioAccountProvider.FactoryMethod,
+                        LogCredentialProviderError);
```

Comment:
You can remove the LogCredentialProviderError from the Importer, since this will now result in double logging (the importer will write the exception message to LogCredentialProviderError, throw, and then TryAddCredentialProvider will also log the exception.